op-node/rollup/driver/state_tob_test.go

```diff
@@ -0,0 +1,253 @@
+func FuzzRejectCreateBlockBadTimestamp(f *testing.F) {
+	f.Fuzz(func(t *testing.T, randSeed int64, l2Time uint64, blockTime uint64, forceOutputFail bool, currentL2HeadTime uint64) {
+		// Create our random provider
+		rng := rand.New(rand.NewSource(randSeed))
+
+		// Create our context for methods to execute under
+		ctx := context.Background()
+
+		// Create our fake L1/L2 heads and link them accordingly
+		l1HeadRef := testutils.RandomBlockRef(rng)
+		l2HeadRef := testutils.RandomL2BlockRef(rng)
+		l2l1OriginBlock := l1HeadRef
+		l2HeadRef.L1Origin = l2l1OriginBlock.ID()
+
+		// TODO: Cap our block time so it doesn't overflow
+		if blockTime > 0x100000 {
+			blockTime = 0x100000
+		}
+
+		// Create a rollup config
+		cfg := rollup.Config{
+			BlockTime: blockTime,
+			Genesis: rollup.Genesis{
+				L1:     l1HeadRef.ID(),
+				L2:     l2HeadRef.ID(),
+				L2Time: l2Time, // dummy value
+			},
+		}
+
+		// Patch our timestamp so we fail
+		l2HeadRef.Time = currentL2HeadTime
+
+		// Create our outputter
+		outputProvider := TestDummyOutputImpl{willError: forceOutputFail}
+
+		// Create our state
+		s := Driver{
+			l1State: &L1State{
+				l1Head:  l1HeadRef,
+				log:     log.New(),
+				metrics: &metrics.Metrics{TransactionsSequencedTotal: prometheus.NewCounter(prometheus.CounterOpts{})},
+			},
+			log:              log.New(),
+			l1OriginSelector: TestDummyL1OriginSelector{retval: l1HeadRef},
+			config:           &cfg,
+			sequencer:        outputProvider,
+			derivation:       TestDummyDerivationPipeline{},
+			metrics:          &metrics.Metrics{TransactionsSequencedTotal: prometheus.NewCounter(prometheus.CounterOpts{})},
+		}
+
+		// Create a new block
+		// - L2Head's L1Origin, its timestamp should be greater than L1 genesis.
+		// - L2Head timestamp + BlockTime should be greater than or equal to the L1 Time.
+		err := s.createNewL2Block(ctx)
+
+		// Verify the L1Origin's timestamp is greater than L1 genesis in our config.
+		if l2l1OriginBlock.Number < s.config.Genesis.L1.Number {
+			assert.Nil(t, err)
+			return
+		}
+
+		// Verify the new L2 block to create will have a time stamp equal or newer than our L1 origin block we derive from.
+		if l2HeadRef.Time+cfg.BlockTime < l2l1OriginBlock.Time {
+			// If not, we expect a specific error.
+			// TODO: This isn't the cleanest, we should construct + compare the whole error message.
+			assert.NotNil(t, err)
+			assert.Contains(t, err.Error(), "cannot build L2 block on top")
+			assert.Contains(t, err.Error(), "for time")
+			assert.Contains(t, err.Error(), "before L1 origin")
+			return
+		}
+
+		// Otherwise we should have no error.
+		assert.Nil(t, err)
+
+		// If we expected the outputter to error, capture that here
+		if outputProvider.willError {
+			assert.NotNil(t, err, "outputInterface failed to createNewBlock, so createNewL2Block should also have failed")
+			return
+		}
+
+		// Otherwise we should have no error.
+		assert.Nil(t, err)
```

Comment:
ditto here: `require` instead of `assert`, and `NoError` instead of `Nil` when checking errors.